### PR TITLE
update IOTA entry to support chrysalis upgrade address format

### PR DESCRIPTION
**Pull request by admljhnsn**
## Description

Updated the [readme.md, added new version of IOTA encoder/decoder, and updated test cases](https://github.com/ensdomains/address-encoder/pull/301/commits/55aa0e635a611d1907279362a197dbdfc356d8f4)

## Reference to the specification

[Updated IOTA address specification for the Chrysalis upgrade](https://chrysalis.docs.iota.org/guides/dev_guide/)

## Reference to the test address.

[IOTA 1.5 address from the IOTA developer specifications](https://chrysalis.docs.iota.org/guides/dev_guide/)
[Second test from the bech32 address format specification](https://github.com/Wollac/protocol-rfcs/blob/bech32-address-format/text/0020-bech32-address-format/0020-bech32-address-format.md)

## List of features added/changed

- made new bech32 encoder/decoder for IOTA based on original one
- test addresses added

NOTE: If you are adding new coin address support, please make sure to add a reference link to README so that reviewer can verify.

## How much has the filesize increased?
Very minimal, updated entries that were already there to new address scheme and added 22 lines in the index.ts

## How Has This Been Tested?

Tested the encoder/decoder for IOTA as in the Usage section of the [readme](https://github.com/ensdomains/address-encoder#usage)

## Checklist:

- [ x] My code follows the code style of this project.
- [ x] My code implements all the required features.
- [ x] I have specified correct coinTypes specified at [Slip 44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
- [ x] I have provided the reference link to the specification I implemented.
- [ x] I have provided enough explanation about how I provided the test address

### CODE REVIEW

**Comment by makoto** [comment]
The link to `Updated IOTA address specification for the Chrysalis upgrade` returns 404.

**Comment by admljhnsn** [comment]
> The link to `Updated IOTA address specification for the Chrysalis upgrade` returns 404.

fixed!

**Comment by makoto** [comment]
What happens to the old format? Are they completely deprecated or do they still support both format?

**Comment by laumair** [comment]
Looks fine to me!

File: src/__tests__/index.test.ts [hunk 1]
```diff
@@ -961,7 +961,7 @@ const vectors: Array<TestVector> = [
     name: 'IOTA',
     coinType: 4218,
     passingVectors: [
-      { text: 'Qc6iYCZWn4BauKXGYirRG8pMtgdHMk2dzn', hex: '3aa9f8f3b055324f6b2d6bcac328ec2d7e3cd22d8b' },
+      { text: 'iota1qpw6k49dedaxrt854rau02talgfshgt0jlm5w8x9nk5ts6f5x5m759nh2ml', hex: '005dab54adcb7a61acf4a8fbc7a97dfa130ba16f97f7471cc59da8b869343537ea' },
```
Comment (Thoralf-M) [inline src/__tests__/index.test.ts:964]: Why is there `00` before the actual address?
It should be `5dab54adcb7a61acf4a8fbc7a97dfa130ba16f97f7471cc59da8b869343537ea` alone I think

**Comment by makoto** [comment]
@laumair thank you for checking. So we don't have to support the old version?

**Comment by antonionardella** [comment]
> @laumair thank you for checking. So we don't have to support the old version?

The old version is only used for the legacy network which is only used to migrate to the new Chrysalis network.

**Comment by laumair** [comment]
> @laumair thank you for checking. So we don't have to support the old version?

@makoto The current state of the old network is that it is still running but it only accepts the migration transactions i.e., a transaction to move your funds from the old network to the new. Migration transactions have a specific format so normal transactions cannot be confirmed on the old / legacy network. 

Personally, I don't see a need to support it here but if you think it would be useful to have it (at least as long as the legacy / old network is active), then perhaps, we could have two versions for IOTA 1) for legacy 2) for mainnet.

File: src/__tests__/index.test.ts (hunk 1 above)
Comment (admljhnsn) [reply src/__tests__/index.test.ts:964]: when I tested the encoder/decoder it output with the '00..' and resolved to the correct 'iota..' address

Comment (Thoralf-M) [reply src/__tests__/index.test.ts:964]: @admljhnsn the nodes will return an error with this added
https://chrysalis-nodes.iota.org/api/v1/addresses/ed25519/005dab54adcb7a61acf4a8fbc7a97dfa130ba16f97f7471cc59da8b869343537ea
`{"error":{"code":"400","message":"invalid parameter, error: invalid address length: 005dab54adcb7a61acf4a8fbc7a97dfa130ba16f97f7471cc59da8b869343537ea: code=400, message=invalid parameter"}}`
https://chrysalis-nodes.iota.org/api/v1/addresses/ed25519/5dab54adcb7a61acf4a8fbc7a97dfa130ba16f97f7471cc59da8b869343537ea
`{"data":{"addressType":0,"address":"5dab54adcb7a61acf4a8fbc7a97dfa130ba16f97f7471cc59da8b869343537ea","balance":0,"dustAllowed":false,"ledgerIndex":1075944}}`

Comment (admljhnsn) [reply src/__tests__/index.test.ts:964]: @Thoralf-M dang, i barely figured out how to get this implemented, i don't know how to fix that here 😢. i guess it has something to do with how the bech32Chain() function works. i'm a dev noob so i don't know what to do

Comment (oopsmonk) [reply src/__tests__/index.test.ts:1008]: The hex(33 bytes) from bech32Chain() is correct, the Chrysalis address consists of version and address parts, it has a leading version byte for ED25519(0x00) and WOTS(0x01). `addresses/ed25519` only accept ed25519 address without the version field. 
see [RFC: Bech32 Address Format](https://github.com/Wollac/protocol-rfcs/blob/bech32-address-format/text/0020-bech32-address-format/0020-bech32-address-format.md)

**Comment by admljhnsn** [comment]
i made some new changes, and don't know the best protocol for going about that in GitHub, so i just edited the original pull request. i hope everything is ok now.

**Review by oopsmonk — APPROVED** [review]

**Comment by makoto** [comment]
Thanks to @admljhnsn @oopsmonk @laumair @Thoralf-M  for your contributions and feedbacks.

**Pull request merged**